### ISSUE TITLE
[BUG] Worldguard flags not working/loading

### ISSUE THREAD

**Issue by cloakfox** [brainsynder-Dev/SimplePets#122]
Please drop the link generated from doing `/pet debug` here
https://www.pastelog.us/soxuferoda

**Describe the bug**
Worldguard flags like allow-pet-spawn, allow-pet-enter & allow-pet-riding are not showing or loading on a new setup, but work fine on an older setup with build 68 (also 1.17.1), but can't get it working with this older build either, logs do not show any errors.

(tried latest build too, and build 68)

**Steps to reproduce**
1. spawn a pet in a worldguard region

**Expected behavior**
To be able to use the worldguard flags

**Screenshots/Videos (If you have any)**
![Capture](https://user-images.githubusercontent.com/3696645/127743049-4ed7d557-9e73-4b02-b27d-29b421287e15.PNG)

**Additional Information**
I have no clue why it works on my other setup, but doesnt on this new one. Im using the same worldedit/worldguard versions on both setups. I've also tried generating a fresh config.yml to see if that fixes it, but it didn't.

**Comment by Thatsmusic99** [comment]
`/pet addon` now contains WorldGuard support rather than have it contained within the plugin itself. Please try installing said addon and reopen the issue if it persists. (However do note the addon does need updating for some flags as it was made when the API was largely incomplete, see https://github.com/BSDevelopers/WorldGuardAddon/issues/1 )

**Comment by cloakfox** [comment]
After installing the Worldguard addon
I still couldn't get it to work, checked bootup log and saw this:
`[10:39:58] [Server thread/INFO]: [SimplePets Error] WorldGuard either wasn't found or isn't enabled.
[10:39:58] [Server thread/INFO]: [SimplePets Error] Please ensure it is installed correctly.`

running the same worldguard version as seen in the debug paste

**Comment by cloakfox** [comment]
> After installing the Worldguard addon
> I still couldn't get it to work, checked console and saw this:
> `[10:39:58] [Server thread/INFO]: [SimplePets Error] WorldGuard either wasn't found or isn't enabled. [10:39:58] [Server thread/INFO]: [SimplePets Error] Please ensure it is installed correctly.`
> 
> running the same worldguard version as seen in the debug paste

after manually enabling it through the in-game command /pet addon
seeing this error spammed in console: https://gist.github.com/cloakfox/6523d1b2eff4611f57e2f6c9e25172de

**Comment by Thatsmusic99** [comment]
oi @brainsynder-Dev update the wg addon kthx

For now, try manually installing it here: https://ci.pluginwiki.us/job/WorldGuardAddon/5/

**Comment by cloakfox** [comment]
> oi @brainsynder-Dev update the wg addon kthx
> 
> For now, try manually installing it here: https://ci.pluginwiki.us/job/WorldGuardAddon/5/

Same issue with that jar file (doesnt detect worldguard)
https://gist.github.com/cloakfox/5bd7c6620aa8e274c31443e0da1e72a5

**Comment by cloakfox** [comment]
> > oi @brainsynder-Dev update the wg addon kthx
> > For now, try manually installing it here: https://ci.pluginwiki.us/job/WorldGuardAddon/5/
> 
> Same issue with that jar file (doesnt detect worldguard)
> https://gist.github.com/cloakfox/5bd7c6620aa8e274c31443e0da1e72a5

Manually loading it with /pets addon command indicates it should work, since there are no errors but it doesnt
https://gist.github.com/cloakfox/e9053aeb162e4a060efc5d8e02a120ea

in-game I still get the denied message "pet was unable to be spawned at the moment"

**Comment by Thatsmusic99** [comment]
> in-game I still get the denied message "pet was unable to be spawned at the moment"

I believe this is WorldGuard's work this time, you need to set `mobs.block-plugin-spawning` in WG's config.yml to true. You will most likely need to restart the server for this to go into effect.

**Comment by cloakfox** [comment]
Currently im on build 84 and it still doesnt automatically enable the worldguard addon, if you set the config value in worldguards config to true it just breaks the entire integration, since it's just ignores worldguard and allows pets everywhere.

We want to allows pets in spawn but disable them in the pvp area.

**Comment by Thatsmusic99** [comment]
Did determine the lack of automatic enabling to be a problem a while ago and should be fixed in [version 0.3 of the WorldGuard addon](https://pluginwiki.us/addons/download/WorldGuard), can you give that a try and let me know if that improves anything?

**Comment by cloakfox** [comment]
> Did determine the lack of automatic enabling to be a problem a while ago and should be fixed in [version 0.3 of the WorldGuard addon](https://pluginwiki.us/addons/download/WorldGuard), can you give that a try and let me know if that improves anything?

The addon loads and the flags appear to be working / loaded on first glance but when I then try to spawn in my pet (in spawn region) it will display the same "Unable to be spawned at the moment" message

This is after i've re-did the worldguard flags to make sure they work.

Is it correct that there is nothing to configure for the worldguard addon? the file is 0 bytes.

**Comment by Thatsmusic99** [comment]
Nothing to configure for the addon, that's correct. Changing the config value now within WorldGuard should work properly now, considering without the addon functioning fully originally, yes, it breaked everything beforehand - but with v0.3 of the addon it should fix it.

If none of that works I'm gonna have to ask you to use an event debugger I built up recently called [ATHENA](https://github.com/Errored-Innovations/ATHENA/releases) and can tell you how to debug the problem.

Figured I'd ask as well, have you got Discord? This may be better to keep track of on there rather than here. If so, feel free to [hop into the support server](https://discord.gg/JEaXvsqzfG), otherwise that's fine.
